### ISSUE TITLE
Application file residing in hdfs

### ISSUE THREAD

**Issue by shyamalshukla** [dotnet/spark#122]
Command : spark-submit --class org.apache.spark.deploy.DotnetRunner --deploy-mode cluster --master spark://<IP>:7077 hdfs://<IP>:9000/user/ContainerAdministrator/NetForSparkDemo/microsoft-spark-2.4.x-0.2.0.jar hdfs://<IP>:9000/user/ContainerAdministrator/NetForSparkDemo/NetForSparkDemo.exe

The driver failed with the below exception. 
Would there be an approximate timeline for this to be available?

Caused by: java.lang.IllegalArgumentException: Launching Python applications through spark-submit is currently only supported for local files: hdfs://<IP>:9000/user/ContainerAdministrator/NetForSparkDemo/NetForSparkDemo.exe

**Comment by imback82** [comment]
There is no timeline set for this. The tricky part to solve here is to figure out all the dependencies of the executable and download them as well, which could be non-trivial. We are open for any suggestion or PR. Thanks!

**Comment by shyamalshukla** [comment]
With Mobius, the application code had to be in hdfs (did not allow file://). 
Are there significant differences with respect to this functionality between spark.net and Mobius?

In the case of local files, how is the same guaranteed?

**Comment by imback82** [comment]
> With Mobius, the application code had to be in hdfs (did not allow file://).
> Are there significant differences with respect to this functionality between spark.net and Mobius?
> 
I don't have too much experience with running spark jobs thru Mobius, so I cannot answer this. @rapoth may help.

> In the case of local files, how is the same guaranteed?
Can you elaborate on this question?

**Comment by rapoth** [comment]
@shyamalshukla : I'm trying to understand your scenario better. Why are you using hdfs?

**Comment by shyamalshukla** [comment]
Yes, I am trying to run the app on a spark cluster. 

From looking at the code in [DotnetRunner.scala](https://github.com/dotnet/spark/blob/master/src/scala/microsoft-spark-2.3.x/src/main/scala/org/apache/spark/deploy/DotnetRunner.scala) recently, I found that the app can be uploaded to hdfs as a zip, which is then downloaded and unzipped by the driver. Would the contents of this archive also be copied to the executor nodes?

**Comment by imback82** [comment]
> Would the contents of this archive also be copied to the executor nodes?

No, the contents will be available only to the driver. If you need to copy something to all of the executor nodes, you need to use either `--jars` or `--files` options built into `spark-submit`.

**Comment by rapoth** [comment]
@shyamalshukla: Are you still blocked on this? Do you need any other help?
